OpenRA.Mods.Common/Widgets/Logic/ModBrowserLogic.cs

```diff
@@ -140,18 +147,46 @@ void RebuildModList()
 			}
 		}
 
-		void SelectMod(ModMetadata mod)
+		void SelectMod(ModMetadata mod, bool? isInstalled)
 		{
+			if (isInstalled == null)
```

Comment:
This is pretty ugly.  You can make this cleaner and also save some unnecessary checks when scrolling through mods by defining a `Cache<string, bool> modInstallStatus`.